solr/core/src/java/org/apache/solr/response/transform/DocTransformer.java

```diff
@@ -41,12 +41,26 @@
    */
   public abstract String getName();
 
+
+  /**
+   * This is called before {@link #transform} to provide context for any subsequent transformations.
+   *
+   * @param context The {@link ResultContext} stores information about how the documents were produced.
+   * @see #needsSolrIndexSearcher
+   */
+  public void prepare( ResultContext context ) {
```

Comment:
I suppose if anything can be done with existing facilities, they should be utilised. I don't take it much if some name is misleading. We have a lot of much more severe issues and live with them.  -1 for "just add"